tests/unit/test_ensemble.py

```diff
@@ -0,0 +1,272 @@
+            ),
+        ]
+    )
+
+
+def test_ensemble(scm_run):
+    ensemble = ScmEnsemble([scm_run, scm_run])
+
+    assert len(ensemble) == len(scm_run) * 2
+    assert ensemble.num_timeseries == len(scm_run) * 2
+    assert ensemble.num_runs == 2
+
+
+def test_ensemble_num_timeseries(ensemble):
+    assert ScmEnsemble().num_timeseries == 0
+
+    assert ensemble.num_timeseries == sum([len(r) for r in ensemble.runs])
+    assert ensemble.num_timeseries > 0
+
+
+def test_ensemble_repr(ensemble):
+    res = repr(ensemble)
+
+    assert "scmdata.ensemble.ScmEnsemble" in res
+    assert "runs: {}".format(ensemble.num_runs) in res
+    assert "timeseries: {}".format(ensemble.num_timeseries) in res
+
+
+def test_ensemble_copy(ensemble):
+    ensemble_copy = ensemble.copy(deep=False)
+    assert id(ensemble) != id(ensemble_copy)
+
+    for l, r in zip(ensemble.runs, ensemble_copy.runs):
+        assert id(l) == id(r)
+
+    ensemble_copy = ensemble.copy(deep=True)
+    assert id(ensemble) != id(ensemble_copy)
+
+    for l, r in zip(ensemble.runs, ensemble_copy.runs):
+        assert id(l) != id(r)
+
+
+@pytest.mark.parametrize("item", ("time", "year", "scenario", "variable"))
+def test_ensemble_getitem(ensemble, item):
+    if item == "year":
+        assert isinstance(ensemble[item], pd.Series)
+        npt.assert_array_almost_equal(ensemble["year"], [2005, 2010, 2015, 2020, 2025])
+    elif item == "time":
+        assert isinstance(ensemble[item], pd.Series)
+        assert np.all(
+            ensemble["time"]
+            == [dt.datetime(y, 1, 1) for y in [2005, 2010, 2015, 2020, 2025]]
+        )
+    else:
+        assert isinstance(ensemble[item], pd.Series)
+        exp = pd.concat([r[item] for r in ensemble.runs])
+        pdt.assert_series_equal(ensemble[item], exp)
+
+
+def test_ensemble_timeseries(ensemble):
+    res = ensemble.timeseries()
+
+    npt.assert_array_almost_equal(ensemble.runs[0]["year"], [2005, 2010, 2015])
+    npt.assert_array_almost_equal(ensemble.runs[1]["year"], [2010, 2015, 2020, 2025])
+    npt.assert_array_almost_equal(res.columns.year, [2005, 2010, 2015, 2020, 2025])
+
+    # index should be in order
+    assert res.index.names == sorted(res.index.names)
+
+    def _get_timeseries(idx, r):
+        ts = r.timeseries()
+        ts["run_id"] = idx
+        ts = ts.set_index("run_id", append=True)
+
+        return ts
+
+    exp = pd.concat([_get_timeseries(run_id, r) for run_id, r in ensemble])
+    exp.index = exp.index.reorder_levels(sorted(exp.index.names))
+
+    pd.testing.assert_frame_equal(res, exp)
+
+    assert np.all(np.isnan(exp.xs(0, level="run_id").loc[:, "2020-01-01":"2025-01-01"]))
+
+    # Check that subset is the same
+    pdt.assert_frame_equal(
+        ensemble.runs[0].timeseries().reset_index(),
+        res.xs(0, level="run_id").dropna(how="all", axis="columns").reset_index(),
+        check_like=True,  # To remove once #97 is resolved
```

Comment:
```suggestion
        check_like=True, 
```

What is #97? Can be removed right?